AceComm-3.0/ChatThrottleLib.lua

```diff
@@ -292,38 +314,89 @@ end
 -- - ... made up of N "Pipe"s (1 for each destination/pipename)
 -- - and each pipe contains messages
 
+local SendAddonMessageResult = Enum.SendAddonMessageResult or {
+	Success = 0,
+	AddonMessageThrottle = 3,
+	NotInGroup = 5,
+	ChannelThrottle = 8,
+	GeneralError = 9,
+}
+
+local function MapToSendResult(result)
+	if result == true then
+		result = SendAddonMessageResult.Success
+	elseif result == false then
+		result = SendAddonMessageResult.GeneralError
+	end
+
+	return result
+end
+
+local function IsThrottledSendResult(result)
+	return result == SendAddonMessageResult.AddonMessageThrottle
```

Comment:
Any particular reason not to check for channelThrottle?

Reply:
Refer to the discussion in the comments: https://github.com/WoWUIDev/Ace3/pull/13#issuecomment-2091476663

ChannelThrottle is neither a new behavior, and that result code being returned still results in the messages sometimes being queued and delivered.

The key word is sometimes - there is an aspect of server-side throttling that's also enforced which may cause them to sometimes be dropped, but this isn't something that's particularly convenient for CTL to track unless we add yet-another ring on a larger cooldown.

Additionally there's a server-side bug that I was informed of that makes the throttling sometimes twice as aggressive as it should be. Either way, if you're hitting this return code then you should assume your messages have a <50% chance of ever making it through.

Reply:
So this boils down to wether its treated as drop and retry or assume delivered question?